### DISCUSSION TITLE
About the distance measurement

### DISCUSSION THREAD

**Discussion by git-liweichao** [microsoft/AirSim#3608] [Support Q&A] [answered]
### What's your question?

Hi, I have a question about the Airsim simulation platform.  I want to measure the distance between the quadcopter and the target on the ground. I think it maybe a slash. I don't know whether the Lidar sensor in airsim can achieve that. If it can, how can I setting? Thanks for you help!

**Comment by ahmed-elsaharti** [comment]
Hi @git-liweichao  Are you trying to get the 'ground truth' distance between the drone and the object or do you want to obtain sensor-based calculation of the distance between the drone and the object?

**Comment by git-liweichao** [comment]
> Hi @git-liweichao Are you trying to get the 'ground truth' distance between the drone and the object or do you want to obtain sensor-based calculation of the distance between the drone and the object?

Thanks for your reply. I want to obtain sensor-based calculation of the distance between the drone and the object. If I can also get the 'ground truth' distance between the drone and the object, it will be more helpful for me.

**Comment by jonyMarino** [comment, marked as answer]
Hi and welcome @git-liweichao! you can accomplish that using `simGetObjectPose` and `simGetVehiclePose` and a subtraction. Take a look at https://microsoft.github.io/AirSim/apis/#common-apis and [simGetVehiclePose](https://microsoft.github.io/AirSim/api_docs/html/index.html?highlight=simgetvehiclepose#airsim.client.VehicleClient.simGetVehiclePose )
